marker_maze.py

```diff
@@ -0,0 +1,242 @@
+                width = abs(corners[m][0][0][0]-corners[m][0][1][0]+corners[m][0][3][0]-corners[m][0][2][0])/2
+                logger.info('marker width %s' % width)
+                if width > self.TURN_WIDTH[self.turn_number]:
+                    self.turn_number += 1
+                    logger.info('Close to marker making turn %s' % self.turn_number)
+                    if self.turn_number == 5:
+                        logger.info('finished!')
+                        self.drive.move(0,0)
+                        self.finished = True
+                pygame.mouse.set_pos(int(found_x), int(self.CROP_WIDTH-found_y))
+                self.t_error = (self.CROP_WIDTH/2 - found_y) / (self.CROP_WIDTH / 2)
+                turn = self.STEERING_OFFSET + self.TURN_P * self.t_error
+                if self.last_t_error is not 0:
+                    #if there was a real error last time then do some damping
+                    turn -= self.TURN_D *(self.last_t_error - self.t_error)
+                turn = min(max(turn,-self.MAX_TURN_SPEED), self.MAX_TURN_SPEED)
+                #if we're rate limiting the turn, go slow
+                if abs(turn) == self.MAX_TURN_SPEED:
+                    self.drive.move (turn, self.STRAIGHT_SPEED/3)
+                else:
+                    self.drive.move (turn, self.STRAIGHT_SPEED)
+                self.last_t_error = self.t_error
+                #print(camera.exposure_speed)
+            else:
+                logger.info("looking for marker %d" % self.turn_number)
+                if self.found:
+                    self.drive.move(0,0)
+                else:
+                    if self.turn_number <= 2:
+                        if self.turn_number == 1:
+                            self.brake()
+                        self.turn_right()
+                    else:
+                        if self.turn_number == 4:
+                            self.brake()
+                        self.turn_left()
+                self.found = False
+                self.last_t_error = 0 
+        else:
+            logger.info("looking for marker %d" % self.turn_number)
+            #if marker was found, then probably best to stop and look
+            if self.found:
+                self.drive.move(0,0)
+            else:
+                #otherwise, go looking
+                if self.turn_number <= 2:
+                    if self.turn_number == 1:
+                        self.brake()
+                    self.turn_right()
+                else:
+                    if self.turn_number == 4:
+                        self.brake()
+                    self.turn_left()
+            self.found = False
+            self.last_t_error = 0
+        # Display the resulting frame
+        frame = pygame.surfarray.make_surface(cv2.flip(frame,1))
+        screen.fill([0,0,0])
+        screen.blit(frame, (0,0))
+        pygame.display.update()
+        if self.found:
+         img_name = str(self.i) + "Fimg.jpg"
```

Comment:
this manual `str()` and concatenation lark is unnecessary - Use string formatting to be more concise (and less error prone):
```
found_identifier = "F" if self.found else "NF"
img_name = "%d%simg.jpg" % (self.i, found_identifier)
```

Reply:
good, thanks. you missed (or haven't yet got to) the code repetition I was expecting you to call me on. there's a nested if, and if either is false, I'd like the same thing to happen, at the moment that's achieved through duplication...